components/director/internal/open_resource_discovery/service.go

```diff
@@ -240,63 +256,197 @@ func (s *Service) processDocuments(ctx context.Context, appID string, baseURL st
+				return nil, errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
 			}
 		}
 		if i, found := searchInSlice(len(apisFromDB), func(i int) bool {
 			return equalStrings(apisFromDB[i].OrdID, &ts.OrdID)
 		}); found {
 			if err := s.apiSvc.Delete(ctx, apisFromDB[i].ID); err != nil {
-				return errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
+				return nil, errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
 			}
 		}
 		if i, found := searchInSlice(len(eventsFromDB), func(i int) bool {
 			return equalStrings(eventsFromDB[i].OrdID, &ts.OrdID)
 		}); found {
 			if err := s.eventSvc.Delete(ctx, eventsFromDB[i].ID); err != nil {
-				return errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
+				return nil, errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
 			}
 		}
 		if i, found := searchInSlice(len(bundlesFromDB), func(i int) bool {
 			return equalStrings(bundlesFromDB[i].OrdID, &ts.OrdID)
 		}); found {
 			if err := s.bundleSvc.Delete(ctx, bundlesFromDB[i].ID); err != nil {
-				return errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
+				return nil, errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
 			}
 		}
 		if i, found := searchInSlice(len(vendorsFromDB), func(i int) bool {
 			return vendorsFromDB[i].OrdID == ts.OrdID
 		}); found {
 			if err := s.vendorSvc.Delete(ctx, vendorsFromDB[i].ID); err != nil {
-				return errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
+				return nil, errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
 			}
 		}
 		if i, found := searchInSlice(len(productsFromDB), func(i int) bool {
 			return productsFromDB[i].OrdID == ts.OrdID
 		}); found {
 			if err := s.productSvc.Delete(ctx, productsFromDB[i].ID); err != nil {
-				return errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
+				return nil, errors.Wrapf(err, "error while deleting resource with ORD ID %q based on its tombstone", ts.OrdID)
+			}
+		}
+		if i, found := searchInSlice(len(fetchRequests), func(i int) bool {
+			return equalStrings(&fetchRequests[i].refObjectOrdID, &ts.OrdID)
+		}); found {
+			fetchRequests = append(fetchRequests[:i], fetchRequests[i+1:]...)
+		}
+
+		for i := range fetchRequests {
+			if equalStrings(&fetchRequests[i].refObjectOrdID, &ts.OrdID) {
+				frIdxToExclude = append(frIdxToExclude, i)
 			}
 		}
 	}
-	return tx.Commit()
+
+	finalFetchRequests := make([]*ordFetchRequest, 0)
+	for i := range fetchRequests {
+		shouldExclude := false
+		for _, idx := range frIdxToExclude {
```

Comment:
the `idx` variable name confused me for a moment, because it is on the position of the value for the entry in `frIdxToExclude`. Maybe consider changing the name. f.e. `idxToExclude`

Reply:
Yes, that's a good idea 👍